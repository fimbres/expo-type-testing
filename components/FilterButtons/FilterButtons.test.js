import React from "react";
import renderer from "react-test-renderer";

import { FilterButtons } from ".";

describe("<FilterButtons />", () => {
  it("render correctly filter different than all", () => {
    const tree = renderer
      .create(
        <FilterButtons
          filter="redemption"
          setFilter={(filter) => console.log(filter)}
        />
      )
      .toJSON();
    expect(tree.children.length).toBe(1);
    expect(tree).toMatchSnapshot();
  });

  it("render correctly filter all", () => {
    const tree = renderer
      .create(
        <FilterButtons
          filter="all"
          setFilter={(filter) => console.log(filter)}
        />
      )
      .toJSON();
    expect(tree.children).toBeUndefined();
    expect(tree).toMatchSnapshot();
  });
});
