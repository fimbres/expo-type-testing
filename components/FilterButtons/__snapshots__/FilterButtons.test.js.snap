// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButtons /> render correctly filter all 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "paddingVertical": 17,
        },
        false,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 12,
            "fontWeight": "800",
            "lineHeight": 16.39,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Ganados
    </Text>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 6,
      }
    }
  />,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "paddingVertical": 17,
        },
        false,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 12,
            "fontWeight": "800",
            "lineHeight": 16.39,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Canjeados
    </Text>
  </View>,
]
`;

exports[`<FilterButtons /> render correctly filter different than all 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "paddingVertical": 14,
      },
      false,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#020202",
          "fontFamily": "avenir-black",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 21.86,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Todos
  </Text>
</View>
`;
